### DISCUSSION TITLE
Escape sharp character

### DISCUSSION THREAD

**Discussion by s-celles** [mermaid-js/mermaid#6104] [Q&A] [answered]
Hello,

I have the following sequence diagram

```mermaid
sequenceDiagram
    participant M as PC (Maître)
    participant E as ATmega16 (Esclave)
    
    rect rgb(200, 255, 200)
        Note right of M: Cas normal
        M->>E: TEMPERATURE\n
        E->>M: VALEUR XXX\n
        Note right of E: XXX = valeur entre -99°C et +999°C
        M->>E: OK\n
    end
    
    rect rgb(255, 220, 220)
        Note right of M: Cas d'erreur
        M->>E: Commande non valide\n
        E->>M: ERROR\n
        Note over M: Recommence<br/>la procédure
    end
    
    Note over E: Retour en attente d'une nouvelle requête
```

I would like to have `#TEMPERATURE\n`, `#VALEUR XXX\n`, `#OK\n` and `#ERROR\n` displayed.

I can't escape sharp character (or I don't know exactly how to proceed.

Some help will be nice.

I ever tried `\#...`, "\#...", '\#...' but none seems to work

Best regards

**Comment by SnowCait** [comment, marked as answer]
It may have already been solved, but for those who encountered the same problem.
`#35;` is the solution.
https://mermaid.js.org/syntax/sequenceDiagram.html#entity-codes-to-escape-characters

**Comment by kreutter** [comment]
Is there a reason why mermaid syntax for escaping characters is slightly different than HTML?
In HTML you would write `&num;` to get #, works in Mermaid with `#num;` **AND** `&num;`
As an alternative in HML you can write `&#35;` to get `#`, but in Mermaid you have to write `#35;` and `&#35;`  will be rendered as `&#`. This is confusing. Especially when you programmatically generate mermaid syntax you are tempted to use a HTML library that does the escaping which obviously doesn't work.
Any reasons for this unnecessary (?) custom syntax?
